Take into account current load when allocating tablets

paszkow opened the pull request:
Currently, tablet allocation intentionally ignores current load (introduced by the commit https://github.com/scylladb/scylladb/commit/1e407ab) which could cause identical shard selection when allocating a small number of tablets in the same topology. When a tablet allocator is asked to allocate N tablets (where N is smaller
than the number of shards on a node), it selects the first N lowest shards. If multiple such tables are created, each allocator run picks the same shards, leading to tablet imbalance across shards.

This change initializes the load sketch with the current shard load, scaled into the [0,1] range, ensuring allocation still remains even while starting from globally least-loaded shards.

Fixes https://github.com/scylladb/scylladb/issues/27620

Needs a backport to all 2025.x versions. In all of them we ignore  the current load.

vponomaryov reviewed (approved):
I confirm that it fixes following bugs:
- https://github.com/scylladb/scylladb/issues/27620
- https://github.com/scylladb/scylladb/issues/27621

I expect this fix to be backported, at least, to the `2025.4`

paszkow commented:
In v3:

- Rebased on top of recently merged sized-based LB and resolved conflicts
- Restored `get_real_avg_shard_load` in load_sketch and used it in NetworkTopologyStrategy::add_tablets_in_racks to correctly choose a node for new tablets
- Fixed `test_imbalance_in_hetero_cluster_with_two_tables` by adding `load_stats.set_default_tablet_sizes(stm.get());` otherwise balancing was never executed due to incomplete tablet stats

@tgrabiec Please, re-review. @nodep Have a look as well. @vponomaryov Let's re-run the test with the fix

tgrabiec commented on locator/load_sketch.hh:
```diff
@@ -99,6 +99,14 @@ class load_sketch {
+        void normalize(load_type factor) {
+            _du.used /= factor;
+            for (shard_id i = 0; i < _shards.size(); ++i) {
+                _shards[i].du.used /= factor;
```
But `network_topology_strategy::add_tablets_in_dc()` is still using `get_avg_tablet_count()`. That path is used for numeric RF. By default we don't go there (determined by rf_rack_valid_keyspaces config option), but since the code is there, we should update it too.

bhalevy commented:
> [test_repair_disjoint_row_3nodes_diff_shard_count](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/23285/testReport/junit/repair_additional_test/TestRepairAdditional/release___dtest_with_consistent_topology_changes___test_repair_disjoint_row_3nodes_diff_shard_count)

https://github.com/scylladb/scylladb/issues/27666

vponomaryov commented:
The latest PR state is being tested here:
- [scylla-staging/valerii/vp-longevity-5000-tables-latte#25](https://argus.scylladb.com/tests/scylla-cluster-tests/a7ef5154-de9f-401c-93f8-9590a10e1107)

At the moment of writing this comment the `data population` phase already finished:
<img width="1570" height="562" alt="Screenshot from 2026-01-06 13-56-23" src="https://github.com/user-attachments/assets/c9f724c1-4c3e-435b-a5a7-786af7140469" />

It has a bit worse perf (`~300kop/s`) than the previous PR state (`~315kop/s`) tested at [scylla-staging/valerii/vp-longevity-5000-tables-latte#23](https://argus.scylladb.com/tests/scylla-cluster-tests/ed7b7497-1713-40b2-afcc-f26f4d51e03d):

<img width="1563" height="453" alt="Screenshot from 2026-01-06 14-06-01" src="https://github.com/user-attachments/assets/7a1c5406-39cc-4fd0-9a37-a6351bebe332" />
